server/src/main/java/org/opensearch/index/engine/EngineConfig.java

```diff
@@ -125,18 +128,21 @@ public Supplier<RetentionLeases> retentionLeasesSupplier() {
      * This setting is also settable on the node and the index level, it's commonly used in hot/cold node archs where index is likely
      * allocated on both `kind` of nodes.
      */
-    public static final Setting<String> INDEX_CODEC_SETTING = new Setting<>("index.codec", "default", s -> {
+    public static final Setting<String> INDEX_CODEC_SETTING = new Setting<>("index.codec", "lz4", s -> {
         switch (s) {
+            case "lz4":
             case "default":
             case "best_compression":
+            case "zlib":
             case "zstd":
             case "zstd_no_dict":
             case "lucene_default":
                 return s;
             default:
                 if (Codec.availableCodecs().contains(s) == false) { // we don't error message the not officially supported ones
                     throw new IllegalArgumentException(
-                        "unknown value for [index.codec] must be one of [default, best_compression, zstd, zstd_no_dict] but was: " + s
+                        "unknown value for [index.codec] must be one of [default, best_compression, lz4, zlib, zstd, zstd_no_dict] but was: "
```

Comment:
Follow up: Can include these values in the docs for index.codec as possible inputs that users can provide.

Reply:
yes, I have added the comment in the PR related to this. We will need to add support to OSB as well.